src/modalities/models/model.py

```diff
@@ -20,6 +20,39 @@ def forward(self, inputs: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
     def get_parameters(self) -> Dict[str, torch.Tensor]:
         return {name: param for name, param in self.named_parameters()}
 
+class SwiGLU(nn.Module):
+    def __init__(self, n_embd: int, bias: bool):
+        super().__init__()
+
+        hidden_dim = self._get_hidden_dim(n_embd)
+
+        self.c_fc = nn.Linear(
+            in_features=n_embd,
+            out_features=hidden_dim,  
+            bias=bias,
+        )
+        self.silu = nn.SiLU()
+        self.c_proj = nn.Linear(
+            in_features=n_embd,
+            out_features=hidden_dim,
+            bias=bias,
+        )
+        self.out_proj = nn.Linear(
+            in_features=hidden_dim,
+            out_features=n_embd,
+            bias=bias,
+        )
+
+    def _get_hidden_dim(self, n_embd: int) -> int:
```

Comment:
we should make this method static

Reply:
addressed in https://github.com/Modalities/modalities/pull/154/commits/4e579b6026b4b0d577892bf68d946af1f8aa545e